From 6a401a03ff6faa4a6ff3a76dcf64680a64c9afe6 Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Wed, 26 Mar 2025 11:35:32 +0100
Subject: [PATCH 2/4] configs/topic_miami*_defconfig: configuration tweaks

The Zynq CPU only has one memory bank
Disable unused EFI partitions and FAT writing
Silence verbose UBI messages at boot
Synchronize the "lite" and "plus" configurations

Signed-off-by: Mike Looijmans <mike.looijmans@topic.nl>
Upstream-Status: Pending
---
 configs/topic_miami_defconfig     |  7 +++++++
 configs/topic_miamilite_defconfig | 25 ++++++++++++++++++++++---
 configs/topic_miamiplus_defconfig | 25 ++++++++++++++++++++++---
 3 files changed, 51 insertions(+), 6 deletions(-)

diff --git a/configs/topic_miami_defconfig b/configs/topic_miami_defconfig
index e736a64f371..06604097088 100644
--- a/configs/topic_miami_defconfig
+++ b/configs/topic_miami_defconfig
@@ -5,6 +5,7 @@ CONFIG_SPL_SYS_DCACHE_OFF=y
 CONFIG_SYS_L2CACHE_OFF=y
 CONFIG_ARCH_ZYNQ=y
 CONFIG_TEXT_BASE=0x4000000
+CONFIG_NR_DRAM_BANKS=1
 CONFIG_SF_DEFAULT_SPEED=108000000
 CONFIG_ENV_SIZE=0x8000
 CONFIG_DM_GPIO=y
@@ -58,6 +59,7 @@ CONFIG_CMD_CACHE=y
 # CONFIG_CMD_EFICONFIG is not set
 CONFIG_CMD_SQUASHFS=y
 CONFIG_CMD_UBI=y
+# CONFIG_EFI_PARTITION is not set
 CONFIG_OF_EMBED=y
 CONFIG_ENV_OVERWRITE=y
 CONFIG_ENV_RELOC_GD_ENV_ADDR=y
@@ -75,6 +77,7 @@ CONFIG_MTD_BLOCK=y
 CONFIG_SPI_FLASH_STMICRO=y
 # CONFIG_SPI_FLASH_USE_4K_SECTORS is not set
 CONFIG_SPI_FLASH_MTD=y
+CONFIG_UBI_SILENCE_MSG=y
 CONFIG_MTD_UBI_BEB_LIMIT=0
 CONFIG_UBI_BLOCK=y
 CONFIG_DEBUG_UART_ZYNQ=y
@@ -90,5 +93,9 @@ CONFIG_USB_GADGET_VENDOR_NUM=0x03fd
 CONFIG_USB_GADGET_PRODUCT_NUM=0x0300
 CONFIG_CI_UDC=y
 CONFIG_USB_GADGET_DOWNLOAD=y
+# CONFIG_FAT_WRITE is not set
+CONFIG_UBIFS_SILENCE_MSG=y
+# CONFIG_UBIFS_SILENCE_DEBUG_DUMP is not set
 CONFIG_SYS_TIMER_COUNTS_DOWN=y
+CONFIG_SPL_CRC32=y
 CONFIG_LZ4=y
diff --git a/configs/topic_miamilite_defconfig b/configs/topic_miamilite_defconfig
index 6c89dd40e26..981d8a96bb0 100644
--- a/configs/topic_miamilite_defconfig
+++ b/configs/topic_miamilite_defconfig
@@ -5,6 +5,7 @@ CONFIG_SPL_SYS_DCACHE_OFF=y
 CONFIG_SYS_L2CACHE_OFF=y
 CONFIG_ARCH_ZYNQ=y
 CONFIG_TEXT_BASE=0x4000000
+CONFIG_NR_DRAM_BANKS=1
 CONFIG_SF_DEFAULT_SPEED=108000000
 CONFIG_ENV_SIZE=0x8000
 CONFIG_DM_GPIO=y
@@ -28,9 +29,9 @@ CONFIG_SYS_CUSTOM_LDSCRIPT=y
 CONFIG_SYS_LDSCRIPT="arch/arm/mach-zynq/u-boot.lds"
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_BOOTDELAY=0
-CONFIG_BOOTCOMMAND="if mmcinfo; then if fatload mmc 0 0x1900000 ${bootscript}; then source 0x1900000; fi; fi; run $modeboot"
 CONFIG_USE_PREBOOT=y
 CONFIG_SYS_PBSIZE=2077
+# CONFIG_BOARD_LATE_INIT is not set
 CONFIG_CLOCKS=y
 CONFIG_SPL_MAX_SIZE=0x30000
 # CONFIG_SPL_SHARES_INIT_SP_ADDR is not set
@@ -41,15 +42,24 @@ CONFIG_SPL_SPI_LOAD=y
 CONFIG_SYS_SPI_U_BOOT_OFFS=0x20000
 CONFIG_SYS_PROMPT="zynq-uboot> "
 CONFIG_SYS_MAXARGS=32
+# CONFIG_CMD_BOOTEFI is not set
+# CONFIG_CMD_ELF is not set
+# CONFIG_CMD_SPL is not set
 CONFIG_CMD_THOR_DOWNLOAD=y
 CONFIG_THOR_RESET_OFF=y
+# CONFIG_CMD_SAVEENV is not set
 CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_DFU=y
 CONFIG_CMD_I2C=y
 CONFIG_CMD_MMC=y
+CONFIG_CMD_MTD=y
 CONFIG_CMD_USB=y
 # CONFIG_CMD_SETEXPR is not set
 CONFIG_CMD_CACHE=y
+# CONFIG_CMD_EFICONFIG is not set
+CONFIG_CMD_SQUASHFS=y
+CONFIG_CMD_UBI=y
+# CONFIG_EFI_PARTITION is not set
 CONFIG_OF_EMBED=y
 CONFIG_ENV_OVERWRITE=y
 CONFIG_ENV_RELOC_GD_ENV_ADDR=y
@@ -57,14 +67,18 @@ CONFIG_NO_NET=y
 CONFIG_SPL_DM_SEQ_ALIAS=y
 CONFIG_DFU_RAM=y
 CONFIG_SYS_DFU_DATA_BUF_SIZE=0x600000
-CONFIG_FPGA_XILINX=y
-CONFIG_FPGA_ZYNQPL=y
 CONFIG_DM_I2C=y
 CONFIG_SYS_I2C_CADENCE=y
 CONFIG_MMC_SDHCI=y
 CONFIG_MMC_SDHCI_ZYNQ=y
+CONFIG_DM_MTD=y
+CONFIG_MTD_BLOCK=y
 CONFIG_SPI_FLASH_STMICRO=y
 # CONFIG_SPI_FLASH_USE_4K_SECTORS is not set
+CONFIG_SPI_FLASH_MTD=y
+CONFIG_UBI_SILENCE_MSG=y
+CONFIG_MTD_UBI_BEB_LIMIT=0
+CONFIG_UBI_BLOCK=y
 CONFIG_DEBUG_UART_ZYNQ=y
 CONFIG_ARM_DCC=y
 CONFIG_ZYNQ_SERIAL=y
@@ -78,4 +92,9 @@ CONFIG_USB_GADGET_VENDOR_NUM=0x03fd
 CONFIG_USB_GADGET_PRODUCT_NUM=0x0300
 CONFIG_CI_UDC=y
 CONFIG_USB_GADGET_DOWNLOAD=y
+# CONFIG_FAT_WRITE is not set
+CONFIG_UBIFS_SILENCE_MSG=y
+# CONFIG_UBIFS_SILENCE_DEBUG_DUMP is not set
 CONFIG_SYS_TIMER_COUNTS_DOWN=y
+CONFIG_SPL_CRC32=y
+CONFIG_LZ4=y
diff --git a/configs/topic_miamiplus_defconfig b/configs/topic_miamiplus_defconfig
index 54849077782..152e0fb647f 100644
--- a/configs/topic_miamiplus_defconfig
+++ b/configs/topic_miamiplus_defconfig
@@ -5,6 +5,7 @@ CONFIG_SPL_SYS_DCACHE_OFF=y
 CONFIG_SYS_L2CACHE_OFF=y
 CONFIG_ARCH_ZYNQ=y
 CONFIG_TEXT_BASE=0x4000000
+CONFIG_NR_DRAM_BANKS=1
 CONFIG_SF_DEFAULT_SPEED=108000000
 CONFIG_ENV_SIZE=0x8000
 CONFIG_DM_GPIO=y
@@ -28,9 +29,9 @@ CONFIG_SYS_CUSTOM_LDSCRIPT=y
 CONFIG_SYS_LDSCRIPT="arch/arm/mach-zynq/u-boot.lds"
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_BOOTDELAY=0
-CONFIG_BOOTCOMMAND="if mmcinfo; then if fatload mmc 0 0x1900000 ${bootscript}; then source 0x1900000; fi; fi; run $modeboot"
 CONFIG_USE_PREBOOT=y
 CONFIG_SYS_PBSIZE=2077
+# CONFIG_BOARD_LATE_INIT is not set
 CONFIG_CLOCKS=y
 CONFIG_SPL_MAX_SIZE=0x30000
 # CONFIG_SPL_SHARES_INIT_SP_ADDR is not set
@@ -41,15 +42,24 @@ CONFIG_SPL_SPI_LOAD=y
 CONFIG_SYS_SPI_U_BOOT_OFFS=0x20000
 CONFIG_SYS_PROMPT="zynq-uboot> "
 CONFIG_SYS_MAXARGS=32
+# CONFIG_CMD_BOOTEFI is not set
+# CONFIG_CMD_ELF is not set
+# CONFIG_CMD_SPL is not set
 CONFIG_CMD_THOR_DOWNLOAD=y
 CONFIG_THOR_RESET_OFF=y
+# CONFIG_CMD_SAVEENV is not set
 CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_DFU=y
 CONFIG_CMD_I2C=y
 CONFIG_CMD_MMC=y
+CONFIG_CMD_MTD=y
 CONFIG_CMD_USB=y
 # CONFIG_CMD_SETEXPR is not set
 CONFIG_CMD_CACHE=y
+# CONFIG_CMD_EFICONFIG is not set
+CONFIG_CMD_SQUASHFS=y
+CONFIG_CMD_UBI=y
+# CONFIG_EFI_PARTITION is not set
 CONFIG_OF_EMBED=y
 CONFIG_ENV_OVERWRITE=y
 CONFIG_ENV_RELOC_GD_ENV_ADDR=y
@@ -57,14 +67,18 @@ CONFIG_NO_NET=y
 CONFIG_SPL_DM_SEQ_ALIAS=y
 CONFIG_DFU_RAM=y
 CONFIG_SYS_DFU_DATA_BUF_SIZE=0x600000
-CONFIG_FPGA_XILINX=y
-CONFIG_FPGA_ZYNQPL=y
 CONFIG_DM_I2C=y
 CONFIG_SYS_I2C_CADENCE=y
 CONFIG_MMC_SDHCI=y
 CONFIG_MMC_SDHCI_ZYNQ=y
+CONFIG_DM_MTD=y
+CONFIG_MTD_BLOCK=y
 CONFIG_SPI_FLASH_STMICRO=y
 # CONFIG_SPI_FLASH_USE_4K_SECTORS is not set
+CONFIG_SPI_FLASH_MTD=y
+CONFIG_UBI_SILENCE_MSG=y
+CONFIG_MTD_UBI_BEB_LIMIT=0
+CONFIG_UBI_BLOCK=y
 CONFIG_DEBUG_UART_ZYNQ=y
 CONFIG_ARM_DCC=y
 CONFIG_ZYNQ_SERIAL=y
@@ -78,4 +92,9 @@ CONFIG_USB_GADGET_VENDOR_NUM=0x03fd
 CONFIG_USB_GADGET_PRODUCT_NUM=0x0300
 CONFIG_CI_UDC=y
 CONFIG_USB_GADGET_DOWNLOAD=y
+# CONFIG_FAT_WRITE is not set
+CONFIG_UBIFS_SILENCE_MSG=y
+# CONFIG_UBIFS_SILENCE_DEBUG_DUMP is not set
 CONFIG_SYS_TIMER_COUNTS_DOWN=y
+CONFIG_SPL_CRC32=y
+CONFIG_LZ4=y
-- 
2.43.0

